scribus/pageitem_textframe.cpp

```diff
@@ -2178,6 +2178,10 @@ void PageItem_TextFrame::layout()
 			// remember possible break
 			if (i + 1 < glyphRuns.count() && glyphRuns[i + 1].hasFlag(ScLayout_LineBoundry))
 			{
+				if((itemText.text(currentIndex -1) != SpecialChars::CJK_NOBREAK_AFTER) &&
+									   (itemText.text(currentIndex) != SpecialChars::CJK_NOBREAK_BEFORE))
+					current.rememberBreak(i-1, breakPos, style.rightMargin());
+
```

Comment:
`currentIndex` is the glyph cluster index not story text index, you should use `current.glyphs[currentIndex - 1].lastChar()` and `current.glyphs[currentIndex].firstChar()`. and make sure `i` is not zero.